docs/tutorials/LSM_single_column_tutorial.jl

```diff
@@ -0,0 +1,386 @@
+# four things:
+# - initialize(land.soil) 
+# - initialize(land.surface_water) 
+# - initializes interaction terms, like p.soil_infiltration
+# - append these into Y, p, and coords:
+Y, p, coords = initialize(land);
+# We have volumetric liquid water fraction:
+propertynames(Y.soil)
+# and surface height of the pond:
+propertynames(Y.surface_water)
+# as well as auxiliary variables for the soil:
+propertynames(p.soil)
+# and nothing for surface water:
+propertynames(p.surface_water)
+# and the shared interaction term
+propertynames(p)
+# and finally, coordinates - useful for visualization of solutions:
+coords.soil
+# and the coordinates of the surface variables:
+coords.surface_water
+
+# And we can make the ode function as before:
+land_ode! = make_ode_function(land);
+
+# Next up would be to set initial conditions, choose a timestepping scheme, and
+# run your simulation.
+
+# # Advantages and disadvantages
+
+# Some advantages to our interface design are as follows:
+# - a developer only needs to learn a few concepts (`rhs!`, prognostic vs. aux variables, `update_aux!`, `initialize`, domains) to make a model which can be run in standalone or work with other components.
+# - likewise, a user only needs to learn one interface to run all models, regardless of if they are standalone components or LSMs with multiple componnents.
+# - the `ode_function! `is completely seperate from the timestepping scheme used, so any scheme can be used (with the exception of mixed implicit/explicit schemes, which we can't handle yet).
+# - although we wrote it here in a ``hardwired`` fashion for surface water and soil, the `update_aux!`, `rhs!`, etc. many methods for LSM models generalize to any number and mix of components. One just needs to write a new model type (e.g. `BiophysicsModel <: AbstractModel` for a vegetation and carbon component model) and the appropriate `interaction` methods for that model.
+# - the order in which the components are treated in the rhs or in update aux does not matter. What matters is that auxiliary/cache variables are updated first, and within this update, interactions are updated last. We assume that the rhs! function for a component only needs the entire `p` and `Y.component` in making this statement. Similarily, updating the aux variables of a single component does not require interaction variables. Yhis is also the same as saying they can be run in *standalone* mode. 
+# - the code is also modular in terms of swapping out a simple component model for a more complex version.
+
+# Some disadvantages to our interface design:
+# - Even in standalone model, variables are accessed in a nested way: Y.soil, p.soil, etc, which is excessive.
+# - To accomodate the fact that some components involve PDEs, a developer for purely ODE based component does need to at least handle `ClimaCore.Field.FieldVector`s.
+# - standalone models need to play by the rules of `AbstractModel`s, and LSMs need to play by the rules of `ClimaLSM.jl`.
```

Comment:
I'm not sure this is a disadvantage, personally :) but perhaps it will feel that way in the short-term.

Reply:
fair enough! yes, maybe the main disadvantage is that there is more machinery to learn up front for standalone models.

changed to "Possible disadvantages" :)